### PR TITLE
Set up Android tests on CI

### PR DESCRIPTION
**Pull request by geeksilva97**
Issue #4369

### Proposed changes

Configures android tests on unix CI

### CODE REVIEW

File: test/test-udp-multicast-join6.c [hunk 1]
```diff
@@ -192,6 +192,11 @@ TEST_IMPL(udp_multicast_join6) {
 
   ASSERT_OK(r);
 
+#if defined(__ANDROID__)
+  /* It returns an ENOSYS error */
+  RETURN_SKIP("Test does not currently work in ANDROID");
+#endif
+
```
Comment (geeksilva97) [inline test/test-udp-multicast-join6.c:199]: This test returns a 38 (ENOSYS) in Android

**Comment by geeksilva97** [comment]
Hello @bnoordhuis . Some of the tests are causing timeout on android. Let me know if we can continue skipping them for a while or if we should spend more time trying to understand what's going on.

### Timeline

Here's the list of tests that we causing timeout

```
tcp_ping_pong
tcp_ping_pong_vec
tcp6_ping_pong
tcp6_ping_pong_vec
tcp_writealot
watcher_cross_stop
```

After skipping the tests above the test `tcp_connect6_link_local` started fail due a timeout (not sure if they are correlate). After skipping it the `pipe` tests like `pipe_ping_pong` started to fail.

I wonder if it's due to the image or something. In my machine I got Android in the same API level and the same ABI and all tests pass.

**Comment by saghul** [comment]
Perhaps it has something to with the emulator running on a VM, making it slower?

Do we have a way to enlarge the timeout with an env variable perhaps?

**Comment by geeksilva97** [comment]
> Perhaps it has something to with the emulator running on a VM, making it slower?
> 
> Do we have a way to enlarge the timeout with an env variable perhaps?

It might be the case. I am trying to find a way to increase the timeout.

**Comment by geeksilva97** [comment]
Is it okay if I switch the docker image?

**Comment by saghul** [comment]
> > Perhaps it has something to with the emulator running on a VM, making it slower?
> 
> > 
> 
> > Do we have a way to enlarge the timeout with an env variable perhaps?
> 
> 
> 
> It might be the case. I am trying to find a way to increase the timeout.

Should be in runner-unix.c iirc.

**Comment by geeksilva97** [comment]
> > > Perhaps it has something to with the emulator running on a VM, making it slower?
> > 
> > 
> > > 
> > 
> > 
> > > Do we have a way to enlarge the timeout with an env variable perhaps?
> > 
> > 
> > It might be the case. I am trying to find a way to increase the timeout.
> 
> Should be in runner-unix.c iirc.

Thank you! I will check it.

**Comment by geeksilva97** [comment]
It worked! 🚀 Ty @saghul .

https://github.com/libuv/libuv/actions/runs/10621295245/job/29442953925?pr=4517

P.S it failed in the sanitizer or something - does not seem to be related to the changes.

File: test/test-emfile.c [hunk 2]
```diff
@@ -75,6 +75,14 @@ TEST_IMPL(emfile) {
   ASSERT_EQ(errno, EMFILE);
   close(maxfd);
 
+#if defined(__ANDROID__)
+  /* Android connect syscall requires an extra file descriptor
```
Comment (bnoordhuis) [inline test/test-emfile.c:79]: Can you elaborate? I'm intrigued :-)

**Review by bnoordhuis — APPROVED** [review]
LGTM. Thanks for doing this!

File: test/test-emfile.c (hunk 2 above)
Comment (geeksilva97) [reply test/test-emfile.c:79]: For sure. I don't think I will have the right terms - I even tried to find more details about it on the internet but I couldn't.

If I understood correctly, this test depletes file descriptors and close on to be used by `uv_tcp_connect`. 

Well, on Android, even though the fd 31 was available it was returning an EMFILE error. After some debugging I realized that [here](https://github.com/libuv/libuv/blob/f00d4b677556a007a28fcb0bb3548ecdb54a29ae/src/unix/tcp.c#L310), `connect` returns `-1` and sets the `errno` even though the `uv__stream_fd` returns the right file descriptor.

Not sure if it's an Android protection or something.

I then closed one more fd and it worked as expected.

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Great work!

**Comment by bnoordhuis** [comment]
Crap, I accidentally clicked 'Update branch' and now there's a merge commit. I guess squash&merge should still work okay.

**Pull request merged**